src/plugins/vis_type_timeseries/public/application/components/lib/convert_series_to_datatable.ts

```diff
@@ -6,51 +6,73 @@
  * Side Public License, v 1.
  */
 import { IndexPattern } from 'src/plugins/data/public';
-import {
-  Datatable,
-  DatatableRow,
-  DatatableColumn,
-  DatatableColumnType,
-} from 'src/plugins/expressions/public';
+import { DatatableRow, DatatableColumn, DatatableColumnType } from 'src/plugins/expressions/public';
+import { Query } from 'src/plugins/data/common';
 import { TimeseriesVisParams } from '../../../types';
 import type { PanelData } from '../../../../common/types';
+import { BUCKET_TYPES } from '../../../../common/enums';
 import { fetchIndexPattern } from '../../../../common/index_patterns_utils';
 import { getDataStart } from '../../../services';
 import { X_ACCESSOR_INDEX } from '../../visualizations/constants';
+import type { TSVBTables } from './types';
 
-interface TSVBTables {
-  [key: string]: Datatable;
+interface FilterParams {
+  filter?: Query;
+  label?: string;
+  field?: string;
 }
 
 interface TSVBColumns {
   id: number;
   name: string;
-  isSplit: boolean;
+  isMetric: boolean;
+  type: string;
+  params?: FilterParams[];
 }
 
 export const addMetaToColumns = (
   columns: TSVBColumns[],
-  indexPattern: IndexPattern,
-  metricsType: string
+  indexPattern: IndexPattern
 ): DatatableColumn[] => {
   return columns.map((column) => {
     const field = indexPattern.getFieldByName(column.name);
     const type = (field?.spec.type as DatatableColumnType) || 'number';
```

Comment:
Not sure whether it matters, but this isn't correct for filters - it will assume type as `number` because there is no field in this case.

Reply:
@flash1293 this is now the case on viz, here is the example of the meta on a filters column (taken by xy axis)
![image](https://user-images.githubusercontent.com/17003240/117956465-be201b80-b321-11eb-8e14-b020bb22a4f3.png)

type === number

Reply:
I see, guess we should fix this on esaggs side as well. Anyway, for consistency reasons it makes sense to keep it for now 👍